### PR TITLE
COMP: Improve top-level completion of macro calls

### PR DESCRIPTION
**Pull request by dima74**
* Added out-of-scope completion for unqualified macro calls
* Support for 3-segments and longer macro calls

output

changelog: Improve top-level completion of macro calls

### CODE REVIEW

File: src/main/kotlin/org/rust/lang/core/completion/RsMacroCompletionProvider.kt [hunk 1]
```diff
@@ -44,31 +42,24 @@ object RsMacroCompletionProvider : RsCompletionProvider() {
...
             .filter { it !is PsiWhiteSpace && it !is PsiComment && it !is PsiErrorElement }
-            .take(4).toList()
-        val is2segmentPath = leftSiblings.getOrNull(0)?.elementType == COLONCOLON &&
-            leftSiblings.getOrNull(1)?.elementType == IDENTIFIER && (
-            leftSiblings.getOrNull(2)?.elementType != COLONCOLON ||
-                leftSiblings.getOrNull(3)?.elementType != IDENTIFIER)
-        val context = RsCompletionContext(isSimplePath = !is2segmentPath)
+            .takeWhile { it.elementType in listOf(IDENTIFIER, COLONCOLON) }
+        if (leftSiblings.drop(MAXIMUM_SUPPORTED_SEGMENTS * 2).any()) return
+        val leftSiblingsText = leftSiblings.toList().reversed().joinToString("") { it.text }
```
Comment (vlad20012) [inline src/main/kotlin/org/rust/lang/core/completion/RsMacroCompletionProvider.kt:51]:
```suggestion
            .takeWhile { it.elementType in listOf(IDENTIFIER, COLONCOLON) }
            .toList()
        if (leftSiblings.size > MAXIMUM_SUPPORTED_SEGMENTS * 2) return
        val leftSiblingsText = leftSiblings.asReversed().joinToString("") { it.text }
```

File: src/main/kotlin/org/rust/lang/core/completion/RsCommonCompletionProvider.kt [hunk 2]
```diff
@@ -172,7 +172,10 @@ object RsCommonCompletionProvider : RsCompletionProvider() {
     ) {
         run {
             // true if delegated from RsPartialMacroArgumentCompletionProvider
-            if (position.containingFile.originalFile is RsExpressionCodeFragment) return
+            val originalFile = position.containingFile.originalFile
```
Comment (vlad20012) [inline src/main/kotlin/org/rust/lang/core/completion/RsCommonCompletionProvider.kt:175]: Let's move the comment one line down.

**Review by vlad20012 — APPROVED** [review]
I like the approach :+1:

**Comment by dima74** [comment]
bors r=vlad20012

**Pull request merged**